### PR TITLE
OperationsGroup is added

### PR DESCRIPTION
**Pull request by DRITE**
and Operations.tsx is refactored

instead of #11

### CODE REVIEW

File: ui/src/components/Operations/Operations.tsx [hunk 1]
```diff
@@ -41,64 +41,22 @@ export function Operations(props: OperationsOwnProps) {
...
-                            >
-                                {trigger}
-                            </Dropdown>
-                        ) : (
-                            trigger
+                        return (
+                            <OperationsGroup
+                                key={group.type}
+                                group={group}
+                                hiddenGroups={hiddenGroups}
```
Comment (Dergash) [inline ui/src/components/Operations/Operations.tsx:51]: `hiddenGroups` are deprecated and shouldn't be present in the demo or archetype

File: ui/src/components/Operations/components/OperationsGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import { ButtonType } from 'antd/lib/button/button'
+import { WidgetTypes } from '@tesler-ui/schema'
+import needRemoveOperation from '../../../utils/needRemoveOperation'
+
+interface OperationsGroupProps {
+    group: OperationGroup
+    onClickCreator: (operation: Operation) => () => void
+    widgetType: WidgetTypes | string
+    hiddenGroups?: string[]
+    buttonType?: ButtonType
```
Comment (Dergash) [inline ui/src/components/Operations/components/OperationsGroup.tsx:15]: It's not assigned anywhere

**Review by Dergash — CHANGES_REQUESTED** [review]

File: ui/src/components/ui/OperationButton/OperationButton.tsx [hunk 3]
```diff
@@ -7,7 +7,11 @@ import { ButtonProps } from 'antd/lib/button/button'
 function OperationButton(props: ButtonProps) {
     const { className, children, onClick, ...rest } = props
     return (
-        <Button {...rest} onClick={onClick} className={cn(styles.operation, className)}>
+        <Button
+            {...rest}
```
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/components/ui/OperationButton/OperationButton.tsx:11]: ok

File: ui/src/components/ui/OperationButton/OperationButton.tsx [hunk 4]
```diff
@@ -7,7 +7,11 @@ import { ButtonProps } from 'antd/lib/button/button'
 function OperationButton(props: ButtonProps) {
```
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/components/ui/OperationButton/OperationButton.tsx:7]: OperationButton is moved to `components/Operations/components/OperationButton.tsx`

File: ui/src/utils/needRemoveOperation.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { OperationScope } from '@tesler-ui/core/interfaces/operation'
+import { WidgetTypes } from '@tesler-ui/schema'
+import { removeRecordOperationWidgets } from '../interfaces/widget'
+
+/**
+ * Check whether the operation with `scope` should be removed
```
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/utils/needRemoveOperation.ts:6]: fixed

[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/utils/needRemoveOperation.ts:6]: > I think needRemoveOperation is very misleading

Why? It was really hard to come up with the name. I've decided to follow next logic:
The function is used inside `if` operator. So what if name would be usable in that conеtext?
`if (needRemoveOperation({ widgetType: widgetType, scope: operation.scope }))` can be read as 'if need to remove operation'
I guess it looks pretty clear =)

> Probably easier to just copypaste this and not make an utility out of this

In my opinion this is a rather complicated condition to use it as is several times. That why I've extracted it to the utility.

File: ui/src/components/Operations/Operations.tsx [hunk 6]
```diff
@@ -41,64 +41,22 @@ export function Operations(props: OperationsOwnProps) {
...
+                                widgetType={widgetMeta.type}
+                                onClickCreator={createClickHandler}
```
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/components/Operations/Operations.tsx:53]: You're right.
`createClickHandler`  is stale name as `Operations` is based on existing component from client project.

File: ui/src/utils/needRemoveOperation.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Dergash) [reply ui/src/utils/needRemoveOperation.ts:6]: It does not say anything about why we need to remove operation and how we gonna do it and it's a global utility completely out of context from where it's used.

Try writing a unit test for this utility - what results should you expect based on the function name and args?

What happens when you need to remove operation for the reason aside of scope?

File: ui/src/components/Operations/Operations.tsx (hunk 1 above)
Comment (DRITE) [reply ui/src/components/Operations/Operations.tsx:51]: It's strange IDE doesn't highlight that so I haven't noticed that. I'll remove it, thank you

File: ui/src/components/Operations/Operations.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/components/Operations/Operations.tsx:53]: fixed

File: ui/src/components/Operations/components/OperationsGroup.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+function OperationsGroup({ group, widgetType, onClickCreator, hiddenGroups, buttonType }: OperationsGroupProps) {
+    if (group.type && hiddenGroups?.includes(group.type)) {
+        return null
+    }
+    let groupIsEmpty = true
```
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/components/Operations/components/OperationsGroup.tsx:22]: fixed

File: ui/src/components/ui/OperationButton/OperationButton.module.css [hunk 8]
```diff
@@ -12,6 +15,10 @@
     margin-right: 0.5em;
 }
 
-.operation, :global(.ant-btn.ant-btn-loading:not(.ant-btn-circle):not(.ant-btn-circle-outline):not(.ant-btn-icon-only)) {
+.base, :global(.ant-btn.ant-btn-loading:not(.ant-btn-circle):not(.ant-btn-circle-outline):not(.ant-btn-icon-only)) {
```
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/components/ui/OperationButton/OperationButton.module.css:18]: It seems redundant. I'll remove it

File: ui/src/components/Operations/components/OperationsGroup.tsx (hunk 2 above)
Comment (DRITE) [reply ui/src/components/Operations/components/OperationsGroup.tsx:15]: it is removed

File: ui/src/components/Operations/components/OperationsGroup.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/components/Operations/components/OperationsGroup.tsx:22]: fixed

File: ui/src/utils/needRemoveOperation.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DRITE) [reply ui/src/utils/needRemoveOperation.ts:6]: > It does not say anything about why we need to remove operation and how we gonna do it and it's a global utility completely out of context from where it's used.

it's equitable

**Review by Dergash — APPROVED** [review]

**Pull request merged**